torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -7164,4 +7165,549 @@ Tensor values_backward(const Tensor& grad, const Tensor& self) {
   return grad_self;
 }
 
+std::tuple<Tensor, Tensor> _betainc_even_partial_numerator(const int32_t iteration, const Tensor& a, const Tensor& b, const Tensor& x) {
+  // Even partial numerator used in the continued fraction for betainc.
+  /*
+   * This function computes the partial numerator d_{2m} that is specified
+   * here: https://dlmf.nist.gov/8.17.E23
+   */
+  auto options = at::TensorOptions().dtype(x.dtype()).device(x.device());
+  const Tensor one = at::scalar_tensor(1.0, options);
+  const Tensor two = at::scalar_tensor(2.0, options);
+  int32_t m = iteration;
+  Tensor a_plus_2m = a + two * m;
+  Tensor a_plus_2m_minus_one = a_plus_2m - one;
+  Tensor denominator = a_plus_2m * a_plus_2m_minus_one;
+
+  Tensor db = m * x / denominator;
+  Tensor value = db * (b - m);
+  Tensor da = -value * (a_plus_2m + a_plus_2m_minus_one) / denominator;
+
+  return std::make_tuple(std::move(value), at::cat({std::move(da), std::move(db)}, -1));
+}
+
+std::tuple<Tensor, Tensor> _betainc_odd_partial_numerator(const int32_t iteration, const Tensor& a, const Tensor& b, const Tensor& x) {
+  // Odd partial numerator used in the continued fraction for betainc.
+  /*
+   * This function computes the partial numerator d_{2m + 1} that is specified
+   * here: https://dlmf.nist.gov/8.17.E23
+   */
+  auto options = at::TensorOptions().dtype(x.dtype()).device(x.device());
+  const Tensor one = at::scalar_tensor(1.0, options);
+  const Tensor two = at::scalar_tensor(2.0, options);
+  int32_t m = iteration;
+  Tensor a_plus_m = a + m;
+  Tensor a_plus_2m = a_plus_m + m;
+  Tensor a_plus_2m_plus_one = a_plus_2m + one;
+  Tensor a_plus_b_plus_m = a_plus_m + b;
+  Tensor denominator = a_plus_2m * a_plus_2m_plus_one;
+
+  Tensor db = -a_plus_m * x / denominator;
+  Tensor value = db * a_plus_b_plus_m;
+  Tensor da = -value * ((a_plus_2m + a_plus_2m_plus_one) / denominator) - x * (two * a_plus_m + b) / denominator;
+
+  return std::make_tuple(std::move(value), at::cat({std::move(da), std::move(db)}, -1));
+}
+
+std::tuple<Tensor, Tensor, Tensor> _betainc_modified_lentz_method(const Tensor& a, const Tensor& b, const Tensor& x, const Tensor& use_continued_fraction) {
+  // Returns the continued fraction for betainc by modified Lentz's method.
+  /*
+   * This function implements the method described in the appendix of [1] for
+   * evaluating continued fractions.
+   * [1] Thompson, Ian J., and A. Ross Barnett.
+   *     Coulomb and Bessel functions of complex arguments and order.
+   *     Journal of Computational Physics 64.2 (1986): 490-509.
+   *     https://www.fresco.org.uk/papers/Thompson-JCP64p490.pdf
+   */
+  // a, b, and x have same dtype
+  std::tuple<Tensor, Tensor> eps_tiny = AT_DISPATCH_FLOATING_TYPES_AND2(
```

Comment:
```suggestion
  auto {eps, tiny} = AT_DISPATCH_FLOATING_TYPES_AND2(
```
This should work to simplify the logic below.